pkg/aws/services/acm/acm.go

```diff
@@ -148,38 +149,38 @@ func (s *Service) DeleteCertificate(domain string) error {
 }
 
 func (s *Service) findCertificateForDomain(domain string) (*string, error) {
-	var existing *acm.ListCertificatesOutput
-	var err error
-
-	// NextToken is the way AWS API performs pagination over results.
-	// If NextToken is not nil, there is another page of results to be requested.
+	certs := getACMCertificates(s.Client)
+	for _, certificate := range certs {
+		if *certificate.DomainName == domain {
+			return certificate.CertificateArn, nil
+		}
+	}
 
-	var nextToken string
+	return nil, nil
+}
 
-	// If existing is nil, means we have to request the very first page of results.
-	for existing == nil || nextToken != "" {
-		if existing != nil && existing.NextToken != nil && *existing.NextToken != "" {
-			nextToken = *existing.NextToken
-		}
-		input := &acm.ListCertificatesInput{}
-		if nextToken != "" {
-			input.NextToken = &nextToken
-		}
-		existing, err = s.Client.ListCertificates(input)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
+func getACMCertificates(acmClient acmiface.ACMAPI) []*acm.CertificateSummary {
+	certs := []*acm.CertificateSummary{}
+	listCertificatesOutput, err := acmClient.ListCertificates(&acm.ListCertificatesInput{
+		MaxItems: aws.Int64(100),
+	})
+	if err != nil {
+		panic(err)
```

Comment:
We should not use panic.

Reply:
Oh, copy paste fail. Please, take a look again.